docs/internal/bimonad.md

```diff
@@ -0,0 +1,353 @@
+# Bimonad tutorial
+
+Bifunctors have become a hot topic in scala since ZIO introduction of a second type parameter.
+It's a great way to manage business errors in your code, but it lacks suitable abstractions.
+The most comprehensive set I've seen lies [here](https://github.com/7mind/izumi/tree/develop/fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio).
+But it's built upon practical usage.
+So let's start building a more principled approach. 
+
+## Monads
+
+Let's talk about the basic definitions. We will use the basic category theory terminology, most of which can be found on [Wikipedia](https://en.wikipedia.org/wiki/Monad_(functional_programming)) and [nLab](https://ncatlab.org/nlab/show/monad).
+We will use the following definitions
+
+### Category 
+Starting from this section we would think of Category as some entity with some additional structure.
+The class mentioned below is something like a Set or Type. Its elements are objects of the category. Since we want the object class to be arbitrary big (bigger than classical Set could be) and we do not need equality\identity relation on it, we will say _Class_
```

Comment:
```suggestion
The class mentioned below is something like a Set or a Type. Its elements are objects of the category. Since we want the object class to be arbitrary big (bigger than the classic Set could be) and we do not need equality/identity relation on it, we will name it _Class_.
```